src/core/async-work-cache.ts

```diff
@@ -50,19 +64,21 @@ export class AsyncWorkCache<K, V> {
     return promise;
   }
 
-  delete (key: K) {
-    this._keyToResultMap.delete(key);
-  }
-
-  clear() {
-    this._keyToResultMap.clear();
-  }
-
-  set(key: K, value: V) {
-    this._keyToResultMap.set(key, Promise.resolve(value));
+  /**
+   * Returns a copy of the underlying map which can be mutated as necessary
+   * and then used to create another cache.
+   */
+  fork(): Map<K, Promise<V>> {
+    return new Map(this._keyToResultMap);
   }
+}
 
-  has(key: K) {
-    return this._keyToResultMap.has(key);
-  }
+/**
+ * A map whose entries may not be deleted.
+ */
+export interface SetOnlyMap<K, V> extends ReadonlyMap<K, V> {
```

Comment:
Moved to `immutable.ts` which is the closest thing to a shared location for this